app/services/import_profile_service.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+class ImportProfileService
+  include Diaspora::Logging
+
+  def import_by_user(user)
+    import_by_files(user.export.current_path, user.exported_photos_file.current_path, user.username)
+  end
+
+  def import_by_files(path_to_profile, path_to_photos, username)
+    if path_to_profile.present?
+      logger.info "Import for profile #{username} at path #{path_to_profile} requested"
+      import_user_profile(path_to_profile, username)
+    end
+
+    user = User.find_by(username: username)
+    return if user.nil?
+
+    if path_to_photos.present?
+      logger.info("Importing photos from import file for '#{username}' from #{path_to_photos}")
+      import_user_photos(user, path_to_photos)
+    end
+    remove_file_references(user)
+  end
+
+  private
+
+  def import_user_profile(path_to_profile, username)
+    return unless path_to_profile.nil? || File.exist?(path_to_profile)
+
+    service = MigrationService.new(path_to_profile, username)
+    logger.info "Start validating user profile #{username}"
+    service.validate
+    logger.info "Start importing user profile for '#{username}'"
+    service.perform!
+    logger.info "Successfully imported profile: #{username}"
+  rescue MigrationService::ArchiveValidationFailed => e
+    logger.error "Errors in the archive found: #{e.message}"
+  rescue MigrationService::MigrationAlreadyExists
+    logger.error "Migration record already exists for the user, can't continue"
+  rescue MigrationService::SelfMigrationNotAllowed
+    logger.error "You can't migrate onto your own account"
+  end
+
+  def import_user_photos(user, path_to_photos)
+    return if path_to_photos.nil?
+
+    return unless File.exist?(path_to_photos)
+
+    uncompressed_photos_folder = unzip_photos_file(path_to_photos)
+    user.posts.find_in_batches do |posts|
+      import_photos_for_posts(posts, uncompressed_photos_folder)
+    end
+  end
+
+  def import_photos_for_posts(posts, source_dir)
+    posts.each do |post|
+      post.photos.each do |photo|
+        uploaded_file = "#{source_dir}/#{photo.remote_photo_name}"
+        next unless File.exist?(uploaded_file) && photo.remote_photo_name.present?
+
+        File.open(uploaded_file) do |file|
+          photo.random_string = File.basename(uploaded_file, ".*")
```

Comment:
Puh.. Maybe the right and reliable handling of photos should be done in another PR?

Reply:
Fine with me, photos are definitely a bigger topic. and it also needs to be split up into two PRs anyway to completely do everything, since parts also need to be backported to 0.7.x